subprojects/test-runner/client/src/main/kotlin/com/avito/runner/listener/ArtifactsTestListener.kt

```diff
@@ -116,6 +104,51 @@ internal class ArtifactsTestListener(
         }
     }
 
+    private fun handleFinishedTest(
+        device: Device,
+        artifactsDir: Result<File>,
+        tempDirectory: Path
+    ): TestResult {
+        return artifactsDir
+            .flatMap { dir ->
+                pullResults(
+                    device = device,
+                    remotePath = dir.toPath(),
+                    destination = tempDirectory
+                )
+            }
+            .fold(
+                onSuccess = { dir ->
+                    TestResult.Complete(dir)
+                },
+                onFailure = { throwable ->
+                    handleIncompleteTest(
+                        result = InfrastructureError.FailOnPullingArtifacts(throwable),
+                        device = device,
+                        tempDirectory = tempDirectory.toFile()
+                    )
+                }
+            )
+    }
+
+    private fun pullResults(device: Device, remotePath: Path, destination: Path): Result<File> {
+        val pullResult = device.pullDir(
+            deviceDir = remotePath,
+            hostDir = destination,
+            validator = reportArtifactsPullValidator
+        )
+        return if (device.api >= 30) {
+            val cleaningResult = device.clearDirectory(remotePath)
+                .map { remotePath.toFile() }
+
+            pullResult.combine(cleaningResult) { pullDir, _ ->
+                pullDir
+            }
```

Comment:
Don't want to switch to new behavior for all APIs.
It costs an additional useless adb invocation - more time and reasons to fail.
`pm clear` do the job in previous behavior.